components/support/ktx/src/main/java/mozilla/components/support/ktx/android/net/Uri.kt

```diff
@@ -17,6 +17,19 @@ import java.io.IOException
 import java.util.UUID
 
 private val commonPrefixes = listOf("www.", "mobile.", "m.")
+private val commonDomain = listOf("wikipedia.org")
+
+/**
+ * Returns the host's common domain if found, else null is returned
+ */
+val Uri.hostWithCommonDomain: String?
+    get() {
+        val host = host ?: return null
+        for (domain in commonDomain) {
+            if (host.endsWith(domain)) return domain
+        }
+        return null
+    }
```

Comment:
I would consider moving this code inside `TippyTopIconPreparer`. It is very specific to that use case and outside of it "common domain" is not really a concept that makes sense. This is specifically crafted for the tippy top sites list :)

Reply:
Sure, I'll do that.  Thanks